mod-clans/src/main/java/gg/solarmc/loader/clans/ClanManager.java

```diff
@@ -45,6 +44,36 @@
     private final Cache<Integer,Clan> clans = Caffeine.newBuilder().expireAfterAccess(Duration.ofMinutes(10)).build();
     private final Cache<Integer,Integer> allianceCache = Caffeine.newBuilder().build();
 
+    //i know i'm not supposed to prematurely optimize, but why not turn two queries into one and improve my sql knowledge at the same time?
+    /**
+     * Gets a clan by the id number of a user
+     * @param transaction the transaction
+     * @param userID the ID of the user
+     * @return empty if no clan was found, or the clan that the user is present in
+     */
+    Optional<Clan> getClanByUser(Transaction transaction, int userID) {
+        var result = transaction.getProperty(DSLContext.class)
+                .select(CLANS_CLAN_MEMBERSHIP.CLAN_ID, CLANS_CLAN_INFO.CLAN_NAME, CLANS_CLAN_INFO.CLAN_LEADER, CLANS_CLAN_INFO.CLAN_KILLS, CLANS_CLAN_INFO.CLAN_DEATHS, CLANS_CLAN_INFO.CLAN_ASSISTS)
+                .from(CLANS_CLAN_MEMBERSHIP)
+                .rightJoin(CLANS_CLAN_INFO).on(CLANS_CLAN_MEMBERSHIP.CLAN_ID.eq(CLANS_CLAN_INFO.CLAN_ID))
+                .where(CLANS_CLAN_MEMBERSHIP.USER_ID.eq(userID))
+                .fetchOne();
+
+        if (result == null) return Optional.empty();
+
+        throw new IllegalStateException(result.toString());
```

Comment:
I think you forgot to finish this?

Reply:
oops just delete it and uncomment the rest of the stuff